pkg/scalers/redis_streams_scaler.go

```diff
@@ -19,22 +20,27 @@ type scaleFactor int8
 const (
 	xPendingFactor scaleFactor = iota + 1
 	xLengthFactor
+	lagFactor
 )
 
 const (
 	// defaults
-	defaultDBIndex       = 0
-	defaultTargetEntries = 5
+	defaultDBIndex             = 0
+	defaultTargetEntries       = 5
+	defaultTargetLag           = 5
+	defaultActivationTargetLag = 0
 
 	// metadata names
-	pendingEntriesCountMetadata = "pendingEntriesCount"
-	streamLengthMetadata        = "streamLength"
-	streamNameMetadata          = "stream"
-	consumerGroupNameMetadata   = "consumerGroup"
-	usernameMetadata            = "username"
-	passwordMetadata            = "password"
-	databaseIndexMetadata       = "databaseIndex"
-	enableTLSMetadata           = "enableTLS"
+	lagMetadata                      = "lagCount"
+	pendingEntriesCountMetadata      = "pendingEntriesCount"
+	streamLengthMetadata             = "streamLength"
+	streamNameMetadata               = "stream"
+	consumerGroupNameMetadata        = "consumerGroup"
+	usernameMetadata                 = "username"
+	passwordMetadata                 = "password"
+	databaseIndexMetadata            = "databaseIndex"
+	enableTLSMetadata                = "enableTLS"
+	activationValueTriggerConfigName = "activationTargetLag"
```

Comment:
An small change, by convention this should be `activationLagCount` (the convention is `activationTHRESHOLDKEYNAME`). could you update it in this repo and in docs as well?

Reply:
Hi Jorge, thanks for the feedback! I've updated the naming convention both here and in the documentation, and it should all be reflected in this PR. It would be great if you could take another look! Thanks!